Simulator/actions/messaging_actions.js

```diff
@@ -0,0 +1,257 @@
+import { eventNames, phases } from '../utils/consts';
+import { validateData, validateSchema } from '../utils/validation.js';
+import * as simulatorActions from './simulator_actions';
+
+// Receiving Message Thunks
+export function receiveMessage(payload) {
+  // Routes received messages to other thunks as necessary
+  return (dispatch) => {
+    try {
+      const { msgData, msgName, props: attrs, version } = JSON.parse(payload.data);
+      if (attrs) {
+        dispatch(simulatorActions.setWdcAttrs(attrs));
+      }
+
+      switch (msgName) {
+        case eventNames.LOG: {
+          dispatch(handleLog(msgData.logMsg));
+          break;
+        }
+
+        case eventNames.LOADED: {
+          dispatch(handleLoaded(version));
+          break;
+        }
+
+        case eventNames.INIT_CB: {
+          dispatch(handleInit());
+          break;
+        }
+
+        case eventNames.SUBMIT: {
+          dispatch(handleSubmit());
+          break;
+        }
+
+        case eventNames.SCHEMA_CB: {
+          const schema = msgData.schema;
+          dispatch(handleSchemaCallback(schema));
+          break;
+        }
+
+        case eventNames.DATA_CB: {
+          const tableName = msgData.tableName;
+          const data = msgData.data;
+          dispatch(handleDataCallback(tableName, data));
+          break;
+        }
+
+        case eventNames.SHUTDOWN_CB: {
+          dispatch(handleShutdownCallback());
+          break;
+        }
+
+        case eventNames.ABORT: {
+          const errorMsg = msgData.errorMsg;
+          dispatch(handleAbort(errorMsg));
+          break;
+        }
+
+        case eventNames.DATA_DONE_CB: {
+          dispatch(handleDataDoneCallback());
+          break;
+        }
+
+        case eventNames.ABORT_AUTH: {
+          const errorMsg = msgData.errorMsg;
+          dispatch(handleAbortForAuth(errorMsg));
+          break;
+        }
+        default: {
+          // The message was not for the simulator, ignore it.
+          break;
+        }
+      }
+    } catch (e) {
+      // The message was not for the simulator, ignore it.
```

Comment:
I thought the same thing when I saw this before, but when I did log the messages I noticed that a lot of extensions use the postMessage api, notably the react and redux developer tools do all their communications using it which flooded the console with errors that weren't useful. I could go either way since it could be nice for debugging and throwing away exceptions feels hacky, but it wasn't the best UX to see 15 errors whenever a rerender happened